src/common/language.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Picks the most preferred of the available languages. Comparison only on
+ * the primary language tag. Region, variant, and script tags are ignored.
+ *
+ * @param {readonly string[]} availableLanguages: Languages to pick from.
+ * @param {readonly string[]} preferredLanguages: Languages to prefer.
+ * @return {string|undefined} matching language, if any.
+ * @example
+ *   matchLanguageTags(['en', 'ru', 'uk'], ['be-BY', 'ru-BY', 'en-UK']) // => 'ru'
+ */
+export function pickPreferredLanguage(availableLanguages, preferredLanguages) {
+  for (const preferredLanguage of preferredLanguages) {
+    const preferredLanguageCode = languageTagCode(preferredLanguage);
+    for (const availableLanguage of availableLanguages) {
+      const availableLanguageCode = languageTagCode(availableLanguage);
+      if (availableLanguageCode === preferredLanguageCode)
+        return availableLanguage;
+    }
+  }
+}
+
+/**
+ * Takes the language tag as per RFC5646 ("uk", "uk-UA", "ru-BY", en-GB").
+ * Returns the primary language-code subtag (lower-case, ISO 639-1).
```

Comment:
[ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) only has 2 chars, we're matching for 3 in the regex
```suggestion
 * Returns the primary language-code subtag (lower-case, ISO 639).
```

Reply:
or does RFC5646 have only 2 and thus the regex should match only 2?

Reply:
Yeah. Very well spotted. I'll fix the comment. We are matching any ISO 639 tag.

```
language      = 2*3ALPHA            ; shortest ISO 639 code
```

Languages that we might want to match now or in the future would most certainly be a part of `ISO 639-1` and have two letter codes. But in theory there could be an `ISO-639` language tag like `ruX`, and we don't want to match that as `ru` hence matching 2-3 characters is necessary. Not supposed to happen in practice since they probably think of this when tagging languages.

Reply:
[amended here](https://github.com/bellingcat/ukraine-timemap/pull/30/commits/3d8dac5b4f8dc3587b9bdc9d7c741aba199fdb70#diff-70963869002d86a47f67898ce491381cb33a0d7af17b9ae6cc2fc0b9d7478a00R24)